src/openMVG/sfm/pipelines/sfm_regions_provider.hpp

```diff
@@ -42,6 +42,12 @@ struct Regions_Provider
     for (Views::const_iterator iter = sfm_data.GetViews().begin();
       iter != sfm_data.GetViews().end() && bContinue; ++iter)
     {
+      if (my_progress_bar.hasBeenCanceled()) 
+      {
+          bContinue = false;
+          continue;
```

Comment:
I think it's enough to mark bContinue as volatile. It won't influence the correctness of the program anyway. Worst case: hasBeenCanceled() gets called a bunch of times.